File: app/src/main/java/com/cv4j/app/fragment/SpitalConvFragment.java
package com.cv4j.app.fragment;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.cv4j.app.R;
import com.cv4j.app.adapter.SpitalConvAdapter;
import com.cv4j.app.app.BaseFragment;
import com.cv4j.app.ui.DividerGridItemDecoration;
import com.cv4j.app.ui.GridRecyclerView;
import com.safframework.injectview.Injector;
import com.safframework.injectview.annotations.InjectView;
import com.safframework.tony.common.utils.Preconditions;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Tony Shen on 2017/3/21.
 */

public class SpitalConvFragment extends BaseFragment {

    @InjectView(R.id.recyclerview)
    GridRecyclerView recyclerview;

    List<String> list = new ArrayList<>();


    static RecyclerView.RecycledViewPool myPool = new RecyclerView.RecycledViewPool();

    static {
        myPool.setMaxRecycledViews(0, 10);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_spital_conv, container, false);
        Injector.injectInto(this, v);

        initData();

        return v;
    }

    private void initData() {

        Resources res = getResources();
        Bitmap bitmap = BitmapFactory.decodeResource(res, R.drawable.test_spital_conv);

        String[] filterNames = res.getStringArray(R.array.spatialConvNames);
        if (Preconditions.isNotBlank(filterNames)) {
            for (String filter:filterNames) {
                list.add(filter);
            }
        }

        GridLayoutManager manager = new GridLayoutManager(mContext, 3);
        manager.setRecycleChildrenOnDetach(true);
        recyclerview.setLayoutManager(manager);
        recyclerview.setAdapter(new SpitalConvAdapter(list,bitmap));
        recyclerview.addItemDecoration(new DividerGridItemDecoration(mContext));
        recyclerview.setRecycledViewPool(myPool);
    }
}

File: app/src/main/java/com/cv4j/app/fragment/SpitalConvFragment.kt
package com.cv4j.app.fragment

import android.content.res.Resources
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.cv4j.app.R
import com.cv4j.app.adapter.SpitalConvAdapter
import com.cv4j.app.app.BaseFragment
import com.cv4j.app.ui.DividerGridItemDecoration
import com.safframework.tony.common.utils.Preconditions
import kotlinx.android.synthetic.main.fragment_spital_conv.*
import java.util.*

/**
 *
 * @FileName:
 *          com.cv4j.app.fragment.SpitalConvFragment
 * @author: Tony Shen
 * @date: 2020-05-04 13:09
 * @version: V1.0 <描述当前版本功能>
 */

class SpitalConvFragment : BaseFragment() {

    private val list: MutableList<String> = ArrayList()

    companion object {
        private val myPool: RecyclerView.RecycledViewPool = RecyclerView.RecycledViewPool()

        init {
            myPool.setMaxRecycledViews(0, 10)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View = inflater.inflate(R.layout.fragment_spital_conv, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initData()
    }

    private fun initData() {
        val res: Resources = resources
        val bitmap = BitmapFactory.decodeResource(res, R.drawable.test_spital_conv)
        val filterNames = res.getStringArray(R.array.spatialConvNames)
        if (Preconditions.isNotBlank(filterNames)) {
            for (filter in filterNames) {
                list.add(filter)
            }
        }
        val manager = GridLayoutManager(mContext, 3)
        manager.recycleChildrenOnDetach = true
        recyclerview.layoutManager = manager
        recyclerview.adapter = SpitalConvAdapter(list, bitmap)
        recyclerview.addItemDecoration(DividerGridItemDecoration(mContext))
        recyclerview.setRecycledViewPool(myPool)
    }
}